pkg/controllers/networkinfo/networkinfo_controller.go

```diff
@@ -190,31 +225,35 @@ func (r *NetworkInfoReconciler) GarbageCollector(cancel chan bool, timeout time.
 			return
 		case <-time.After(timeout):
 		}
+		// read all nsx-vpc from vpc store
 		nsxVPCList := r.Service.ListVPC()
 		if len(nsxVPCList) == 0 {
 			continue
 		}
 
-		networkInfoList := &v1alpha1.NetworkInfoList{}
-		err := r.Client.List(ctx, networkInfoList)
+		// read all namespaces from k8s
+		namespaces := &corev1.NamespaceList{}
+		err := r.Client.List(ctx, namespaces)
 		if err != nil {
-			log.Error(err, "failed to list NetworkInfo CR")
+			log.Error(err, "failed to list k8s namespaces")
 			continue
 		}
 
-		crdNetworkInfoVPCStateSet := sets.NewString()
-		for _, networkInfo := range networkInfoList.Items {
-			for _, vpcState := range networkInfo.VPCs {
-				crdNetworkInfoVPCStateSet.Insert(vpcState.VPCPath)
-			}
+		nsSet := sets.NewString()
+		for _, ns := range namespaces.Items {
+			nsSet.Insert(ns.Namespace)
 		}
 
 		for _, elem := range nsxVPCList {
-			if crdNetworkInfoVPCStateSet.Has(*elem.Path) {
+			// for go lint Implicit memory aliasing in for loop
+			// this limitation is fixed after golang 1.22, should remove the temp var after upgrading to 1.22
+			tempElem := elem
+			nsxVPCNamespace := getNamespaceFromNSXVPC(&tempElem)
+			if nsSet.Has(nsxVPCNamespace) {
```

Comment:
I think for NSX Operator restart case, Namespace cannot be deleted as we have the finalizer. So it's better we also check whether Namespace is in deletion?

Reply:
I think this case need to be handled in NetworkInfo CRD reconciler, not in gc logic